nixos/modules/security/dhparams.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.security.dhparams;
+  dir = "/var/cache/dhparams";
+in
+{
+  options = {
+    security.dhparams = mkOption {
+      description =
+        ''
+          Diffie-Hellman parameters to generate.
+
+          The value is the size (in bits) of the DH params to generate. The
+          generated DH params path can be found in `${dir}/[name].pem`
```

Comment:
Could `dir` be configurable or at least exposed somehow? I'd rather not hard-code this path but use something like `${security.dhparams.name.path}` in my nginx config.